File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n == 0) return 0;
        if (n <= 2) return 1;
        int x = 0, y = 1, z = 1;
        n -= 2;
        while (n--) {
            int temp = z + y + x;
            x = y;
            y = z;
            z = temp;
        }
        return z;
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.go
var mem [40]int

func tribonacci(n int) int {
    if n == 0 {
        return 0
    } else if n < 3 {
        return 1
    } else if mem[n] != 0 {
        return mem[n]
    } else {
        mem[n] = tribonacci(n - 1) + tribonacci(n - 2) + tribonacci(n - 3)
        return mem[n]
    }
}